internal/auth/ldap/repository_account.go

```diff
@@ -149,17 +149,28 @@ func (r *Repository) ListAccounts(ctx context.Context, withAuthMethodId string,
 }
 
 // ListDeletedIds lists the public IDs of any accounts deleted since the timestamp provided.
-func (r *Repository) ListDeletedIds(ctx context.Context, since time.Time) ([]string, error) {
+func (r *Repository) ListDeletedIds(ctx context.Context, since time.Time) ([]string, time.Time, error) {
```

Comment:
Renaming of this method will happen in https://github.com/hashicorp/boundary/pull/3777, which applies the patterns in https://github.com/hashicorp/boundary/pull/3761. I agree that the struct is a nice idea!

Reply:
Unfortunately, the use of a struct for the return type here will complicate the use of this method across many services as it's not clear where we would define the type such that it could be used in the `pagination` package interface using this method (through an interface). I'm not sure the added complexity needed to allow the struct is worth it. Given the conversation today, we should only need one timestamp return - the transaction timestamp minus some duration based on the "default request timeout". I'll leave this PR up for that buffer to be added.